torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,467 @@
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(os.path.join(self.root, source, f"{filename}.tif")) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: Optional[str] = None) -> None:
+        """Verify the integrity of the dataset.
+
+        Args:
+            url: url to the file
+            md5: md5 of the file to be verified
+
+        Raises:
+            RuntimeError: if dataset is not found
+        """
+        filepath = os.path.join(self.root, url.split("/")[-1])
+        if os.path.isfile(filepath) and filepath.endswith(".zip"):
+            if self.checksum and not check_integrity(filepath, md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            self._extract(url)
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        elif self.download and url.endswith(".csv"):
+            # Download the split file
+            download_url(url, self.root, filename=os.path.split(url)[1], md5=None)
+
+        elif self.download and not url.endswith(".csv"):
+            # Download the dataset
+            self._download(url, md5)
+            self._extract(url)
+
+    def _download(self, url: str, md5: Optional[str]) -> None:
+        """Download the dataset."""
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, url: str) -> None:
+        """Extract the dataset."""
+        filepath = os.path.join(self.root, os.path.split(url)[1])
+        extract_archive(filepath)
+
+    def _merge_parts(self, root: str, modality: str) -> None:
+        """Merge the modalities that are downloaded and extracted in two parts.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the filename of the modality
+        """
+        fname_p1 = modality + "_part1"
+        fname_p2 = modality + "_part2"
+        source_folder = os.path.join(root, fname_p1)
+        destination_folder = os.path.join(root, fname_p2)
+        for file_name in os.listdir(source_folder):
+            source = os.path.join(source_folder, file_name)
+            destination = os.path.join(destination_folder, file_name)
+            if os.path.isfile(source):
+                shutil.move(source, destination)
+
+        shutil.rmtree(source_folder)
```

Comment:
Would this result in the dataset having to be re-downloaded after every time it is used?

Reply:
The source folder (i.e., s1_part1) has been moved to the destination folder (i.e., s1_part2) before the deletion of source folder. Then, we rename the destination folder like destination_folder.split('_')[0]. So yeah, it is not removing any data.

Reply:
As long as the dataset doesn't try to redownload the files because source folder is empty